src/gluonts/model/deepvar_hierarchical/_estimator.py

```diff
@@ -0,0 +1,308 @@
+
+
+class DeepVARHierarchicalEstimator(DeepVAREstimator):
+    """
+    Constructs a DeepVARHierarchical estimator, which is a hierachical extension of DeepVAR.
+
+    The model has been described in the ICML 2021 paper:
+    http://proceedings.mlr.press/v139/rangapuram21a.html
+
+
+    Parameters
+    ----------
+    freq
+        Frequency of the data to train on and predict
+    prediction_length
+        Length of the prediction horizon
+    target_dim
+        Dimensionality of the input dataset
+    S
+        Summation or aggregation matrix.
+    num_samples_for_loss
+        Number of samples to draw from the predicted distribution to compute the training loss.
+    likelihood_weight
+        Weight for the negative log-likelihood loss. Default: 0.0.
+        If not zero, then negative log-likelihood (times `likelihood_weight`) is added to the default CRPS loss.
+    CRPS_weight
+        Weight for the CRPS loss component. Default: 1.0.
+        If zero, then loss is only negative log-likelihood (times `likelihood_weight`).
+        If non-zero, then loss is CRPS (times 'CRPS_weight') added to the default negative log-likelihood loss
+    sample_LH
+        Boolean flag to switch between likelihoods from samples or NN parameters. Default: False
+    assert_reconciliation
+        Flag to indicate whether to assert if the (projected) samples generated during prediction are coherent.
+    coherent_train_samples
+        Flag to indicate whether sampling/projection is being done during training
+    coherent_pred_samples
+        Flag to indicate whether sampling/projection is being done during prediction
+    trainer
+        Trainer object to be used (default: Trainer())
+    context_length
+        Number of steps to unroll the RNN for before computing predictions
+        (default: None, in which case context_length = prediction_length)
+    num_layers
+        Number of RNN layers (default: 2)
+    num_cells
+        Number of RNN cells for each layer (default: 40)
+    cell_type
+        Type of recurrent cells to use (available: 'lstm' or 'gru';
+        default: 'lstm')
+    num_parallel_samples
+        Number of evaluation samples per time series to increase parallelism
+        during inference. This is a model optimization that does not affect
+        the accuracy (default: 100)
+    dropout_rate
+        Dropout regularization parameter (default: 0.1)
+    cardinality
+        Number of values of each categorical feature (default: [1])
+    embedding_dimension
+        Dimension of the embeddings for categorical features
+        (default: 5])
+    distr_output
+        Distribution to use to evaluate observations and sample predictions
+        (default: LowrankMultivariateGaussianOutput with dim=target_dim and
+        rank=0, i.e., covariance matrix is assumed to be diagonal).
+        Note that target dim of the DistributionOutput and the estimator constructor
+        call need to match.
+    rank
```

Comment:
Should we print a warning to inform the user of unused parameters? I think this is an issue that we have a lot in GluonTS and the docstring can be easily missed.

Reply:
I removed unused parameters from the constructor arguments and explicitly set them at the beginning of constructor. If the method is extended, if at all, then they can be moved into arguments.